### PR TITLE
feat(shuttle-serenity): make serenity 0.12 default, support poise 0.6, deprecate shuttle-poise

### PR DESCRIPTION
**Pull request by supleed2**
## Description of change

- Adds `poise` and `poise-0-6` feature flags, with `poise` enabled by default
- Note similar to `shuttle-serenity` added to readme

## How has this been tested? (if applicable)

- `cargo clippy`
- `cargo clippy --no-default-features -F poise-0-6`

### CODE REVIEW

**Comment by jonaro00** [comment]
Thanks for the PR! :D
I think it'll soon be time for switching the serenity plugin to use 0.12 by default (a PR that you can do if you want to), so I think having this released at the same time with 0.6 as the default feature will be good.

**Comment by supleed2** [comment]
Sounds great, all that's needed for that is changing the defaults? So:

`default = ["poise"]` -> `default = ["poise-0-6"]`
`default = ["rustls_backend"]` -> `default = ["serenity-0-12-rustls_backend"]`

**Comment by jonaro00** [comment]
Rather with a new feature for the older version, and no feature for the current one. See the shuttle-axum crate.

**Comment by jonaro00** [comment]
Since the changes are so closely related, I don't mind having both changes in one PR. Either way works.

**Review by jonaro00 — APPROVED** [review]

File: services/shuttle-poise/src/lib.rs [hunk 1]
```diff
@@ -12,6 +23,16 @@ where
...
 {
     async fn bind(mut self, _addr: SocketAddr) -> Result<(), shuttle_runtime::Error> {
+        #[cfg(feature = "poise")]
+        serenity::ClientBuilder::new(self.token, self.intents)
+            .framework(self.framework)
+            .await
+            .map_err(shuttle_runtime::CustomError::new)?
+            .start_autosharded()
+            .await
+            .map_err(shuttle_runtime::CustomError::new)?;
```
Comment (jonaro00) [inline services/shuttle-poise/src/lib.rs:33]: When poise 0.5 has aged, we can consider having the poise integration just be a part of shuttle-serenity. If I'm not mistaken, you would just make a serenity client, add the poise framework, and then use it just like plain shuttle-serenity.

Comment (supleed2) [reply services/shuttle-poise/src/lib.rs:33]: I didn't realise this but yeah, it totally should! Would be nice to significantly clean things up.

Comment (jonaro00) [reply services/shuttle-poise/src/lib.rs:33]: Hmm... why wait for poise 0.5 to age? We can simply deprecate shuttle-poise in the state before this PR, but with a deprecation warning and a small example on how to use shuttle-serenity + 0.12 + poise 0.6. Does this sound good @oddgrd ?

Comment (oddgrd) [reply services/shuttle-poise/src/lib.rs:33]: Sounds great to me!

**Comment by jonaro00** [comment]
@supleed2 You can restore shuttle-poise to its state before this PR, change the requirement on `shuttle-runtime` to `<=0.40.0`, add a `#[deprecated = "..." ]` on the PoiseService. The readme and the deprecation message can have a link to the latest docs.rs/shuttle-serenity.

Then there is also an opportunity to update the examples to use the newer versions of serenity and poise. :wink:

**Review by jonaro00 — APPROVED** [review]

**Pull request merged**